stan/math/rev/core/vari.hpp

```diff
@@ -68,37 +100,32 @@ class vari {
   }
 
   /**
-   * Throw an illegal argument exception.
-   *
-   * <i>Warning</i>: Destructors should never called for var objects.
-   *
-   * @throw Logic exception always.
+   * Constructor from vari_value
```

Comment:
I've thought about this and I think the answer is no because a `move` is changing the ownership of memory, but I think we always want the memory for a vari to completely reside in the stack. Even with Eigen types we will make memory in the stack allocator then for an input Eigen copy it's memory over to the memory allocated on the stack. Since we always want to create memory for vari on the stack we don't want to accept ownership of any other memory outside of the stack

Reply:
side note: I ran a big test a while ago to see if move semantics and perfect forwarding would be useful for double and var matrices. For double the answer is yes by like 20% but for var the answer is no not at all because of the above

Reply:
> we don't want to accept ownership of any other memory outside of the stack

Oh, well we'd never allocate vari memory anywhere but the stack. I guess without a use for it there's no reason to have it. I don't know of anywhere we actually handle varis as non-pointers.

Reply:
>  I don't know of anywhere we actually handle varis as non-pointers.

Yeah idt we do